pallets/creditcoin/src/types/platform.rs

```diff
@@ -20,38 +36,100 @@ impl From<u64> for EvmChainId {
 }
 
 impl EvmChainId {
-	pub fn new(value: u64) -> Self {
+	pub const fn new(value: u64) -> Self {
 		EvmChainId(value)
 	}
+	pub fn as_u64(self) -> u64 {
```

Comment:
The reason I didn't use `Into` is that it leads to type inference issues, which force you to make a new `let` binding with a type annotation.

For instance for the usages of `as_u64`:
```rust
let id = rpc::eth_chain_id(rpc_url)?.as_u64();
if id == expected.as_u64()
```
and
```rust
let s = chain_id.as_u64().to_string();
```
if you were to use `into()` instead, it would fail to compile because the target type is ambiguous. So you'd have to do something like
```rust
let chain_id_num: u64 = chain_id.into();
let s = chain_id_num.to_string();
```
to resolve the ambiguity, which I find kind of annoying.

I can add an `Into` impl though, it just wouldn't be used anywhere currently